NoClassDefFoundError: AnalyticsConnector

### Actual behaviour
- normal execution of app

### Expected behaviour
************ CAUSE OF ERROR ************

java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/firebase/analytics/connector/AnalyticsConnector;
	at com.google.firebase.messaging.MessagingAnalytics.zza(Unknown Source)
	at com.google.firebase.messaging.MessagingAnalytics.logNotificationReceived(Unknown Source)
	at com.google.firebase.messaging.FirebaseMessagingService.zzd(Unknown Source)
	at com.google.firebase.iid.zzc.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
	at com.google.android.gms.common.util.concurrent.zza.run(Unknown Source)
	at java.lang.Thread.run(Thread.java:831)
Caused by: java.lang.ClassNotFoundException: Didn't find class "com.google.firebase.analytics.connector.AnalyticsConnector" on path: DexPathList[[zip file "/data/app/com.nextcloud.client-1/base.apk"],nativeLibraryDirectories=[/data/app/com.nextcloud.client-1/lib/arm64, /vendor/lib64, /system/lib64]]
	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:511)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:469)
	... 8 more
	Suppressed: java.lang.ClassNotFoundException: com.google.firebase.analytics.connector.AnalyticsConnector
		at java.lang.Class.classForName(Native Method)
		at java.lang.BootClassLoader.findClass(ClassLoader.java:781)
		at java.lang.BootClassLoader.loadClass(ClassLoader.java:841)
		at java.lang.ClassLoader.loadClass(ClassLoader.java:504)
		... 9 more
	Caused by: java.lang.NoClassDefFoundError: Class not found using the boot class loader; no stack available

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30080190
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: Huawei
Device: hwALE-H
Model: ALE-L21
Id: HuaweiALE-L21
Product: ALE-L21

************ FIRMWARE ************
SDK: 21
Release: 5.0.1
Incremental: C432B20

Having this happen to me too, almost the exact same backtrace too with the newest version of the app.

```
************ CAUSE OF ERROR ************

java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/firebase/analytics/connector/AnalyticsConnector;
	at com.google.firebase.messaging.MessagingAnalytics.zza(Unknown Source:68)
	at com.google.firebase.messaging.MessagingAnalytics.logNotificationReceived(Unknown Source:2)
	at com.google.firebase.messaging.FirebaseMessagingService.zzd(Unknown Source:50)
	at com.google.firebase.iid.zzg.run(Unknown Source:4)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at com.google.android.gms.common.util.concurrent.zza.run(Unknown Source:7)
	at java.lang.Thread.run(Thread.java:764)
Caused by: java.lang.ClassNotFoundException: Didn't find class "com.google.firebase.analytics.connector.AnalyticsConnector" on path: DexPathList[[zip file "/system/framework/org.apache.http.legacy.boot.jar", zip file "/data/app/com.nextcloud.client-LTiS90daWMmAsD_F7QRCGQ==/base.apk"],nativeLibraryDirectories=[/data/app/com.nextcloud.client-LTiS90daWMmAsD_F7QRCGQ==/lib/arm64, /data/app/com.nextcloud.client-LTiS90daWMmAsD_F7QRCGQ==/base.apk!/lib/arm64-v8a, /system/lib64]]
	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:134)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
	... 8 more

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30100051
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: asus
Device: ASUS_X00QD
Model: ASUS_X00QD
Id: PPR1.180610.009
Product: WW_X00QD

************ FIRMWARE ************
SDK: 28
Release: 9
Incremental: 16.0615.1908.109-0
```

Seems to often happen during the night for me, with the error message greeting me many mornings, as well as occasionally sometimes during the day when I unlock my phone.

This happens to me too. I have:
- Google Nexus 6P
- OS: LineageOS (Android 8.1)
- Open-GApps nano
- Magisk

```
************ CAUSE OF ERROR ************

java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/firebase/analytics/connector/AnalyticsConnector;
	at com.google.firebase.messaging.MessagingAnalytics.zza(Unknown Source:68)
	at com.google.firebase.messaging.MessagingAnalytics.logNotificationReceived(Unknown Source:2)
	at com.google.firebase.messaging.FirebaseMessagingService.zzd(Unknown Source:50)
	at com.google.firebase.iid.zzg.run(Unknown Source:4)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
	at com.google.android.gms.common.util.concurrent.zza.run(Unknown Source:7)
	at java.lang.Thread.run(Thread.java:764)
Caused by: java.lang.ClassNotFoundException: Didn't find class "com.google.firebase.analytics.connector.AnalyticsConnector" on path: DexPathList[[zip file "/data/app/com.nextcloud.client-XIV3hreoVJQb7KqdFkoWdA==/base.apk"],nativeLibraryDirectories=[/data/app/com.nextcloud.client-XIV3hreoVJQb7KqdFkoWdA==/lib/arm64, /data/app/com.nextcloud.client-XIV3hreoVJQb7KqdFkoWdA==/base.apk!/lib/arm64-v8a, /system/lib64, /vendor/lib64]]
	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:125)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
	... 8 more

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30100190
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: google
Device: angler
Model: Nexus 6P
Id: OPM7.181205.001
Product: angler

************ FIRMWARE ************
SDK: 27
Release: 8.1.0
Incremental: fef9954632
```

Same here:

```
************ CAUSE OF ERROR ************

java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/firebase/analytics/connector/AnalyticsConnector;
 at com.google.firebase.messaging.MessagingAnalytics.zza(Unknown Source:68)
 at com.google.firebase.messaging.MessagingAnalytics.logNotificationReceived(Unknown Source:2)
 at com.google.firebase.messaging.FirebaseMessagingService.zzd(Unknown Source:50)
 at com.google.firebase.iid.zzg.run(Unknown Source:4)
 at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
 at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
 at com.google.android.gms.common.util.concurrent.zza.run(Unknown Source:7)
 at java.lang.Thread.run(Thread.java:764)
Caused by: java.lang.ClassNotFoundException: Didn't find class "com.google.firebase.analytics.connector.AnalyticsConnector" on path: DexPathList[[zip file "/system/framework/org.apache.http.legacy.boot.jar", zip file "/data/app/com.nextcloud.client-3J5837-wQ6ZBQIaMdCT0-Q==/base.apk"],nativeLibraryDirectories=[/data/app/com.nextcloud.client-3J5837-wQ6ZBQIaMdCT0-Q==/lib/arm64, /data/app/com.nextcloud.client-3J5837-wQ6ZBQIaMdCT0-Q==/base.apk!/lib/arm64-v8a, /system/lib64, /system/vendor/lib64]]
 at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:134)
 at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
 at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
 ... 8 more

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30100190
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: samsung
Device: dreamlte
Model: SM-G950F
Id: PPR1.180610.011
Product: dreamltexx

************ FIRMWARE ************
SDK: 28
Release: 9
Incremental: G950FXXS6DTA1
```

Happens to me, too. 
Just happens at random on notification prompts or shortly thereafter.


---------------------
************ CAUSE OF ERROR ************

java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/firebase/analytics/connector/AnalyticsConnector;
	at com.google.firebase.messaging.zzo.zza(com.google.firebase:firebase-messaging@@20.1.3:120)
	at com.google.firebase.messaging.zzo.zza(com.google.firebase:firebase-messaging@@20.1.3:1)
	at com.google.firebase.messaging.FirebaseMessagingService.zzc(com.google.firebase:firebase-messaging@@20.1.3:50)
	at com.google.firebase.messaging.zze.run(com.google.firebase:firebase-messaging@@20.1.3:2)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at com.google.android.gms.common.util.concurrent.zza.run(Unknown Source:6)
	at java.lang.Thread.run(Thread.java:764)
Caused by: java.lang.ClassNotFoundException: Didn't find class "com.google.firebase.analytics.connector.AnalyticsConnector" on path: DexPathList[[zip file "/system/framework/org.apache.http.legacy.boot.jar", zip file "/data/app/com.nextcloud.client-27VxS6aQ91yFJk_114JNPg==/base.apk"],nativeLibraryDirectories=[/data/app/com.nextcloud.client-27VxS6aQ91yFJk_114JNPg==/lib/arm64, /data/app/com.nextcloud.client-27VxS6aQ91yFJk_114JNPg==/base.apk!/lib/arm64-v8a, /system/lib64]]
	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:134)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
	... 8 more

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30110090
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: OnePlus
Device: OnePlus5T
Model: ONEPLUS A5010
Id: PKQ1.180716.001
Product: OnePlus5T

************ FIRMWARE ************
SDK: 28
Release: 9
Incremental: 2002242012

************ CAUSE OF ERROR ************

java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/firebase/analytics/connector/AnalyticsConnector;
	at com.google.firebase.messaging.zzo.zza(com.google.firebase:firebase-messaging@@20.1.3:120)
	at com.google.firebase.messaging.zzo.zza(com.google.firebase:firebase-messaging@@20.1.3:1)
	at com.google.firebase.messaging.FirebaseMessagingService.zzc(com.google.firebase:firebase-messaging@@20.1.3:50)
	at com.google.firebase.messaging.zze.run(com.google.firebase:firebase-messaging@@20.1.3:2)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at com.google.android.gms.common.util.concurrent.zza.run(Unknown Source:6)
	at java.lang.Thread.run(Thread.java:764)
Caused by: java.lang.ClassNotFoundException: Didn't find class "com.google.firebase.analytics.connector.AnalyticsConnector" on path: DexPathList[[zip file "/system/framework/org.apache.http.legacy.boot.jar", zip file "/data/app/com.nextcloud.client-AB0yY_gNZ-P3uHm8tDJPNA==/base.apk"],nativeLibraryDirectories=[/data/app/com.nextcloud.client-AB0yY_gNZ-P3uHm8tDJPNA==/lib/arm64, /data/app/com.nextcloud.client-AB0yY_gNZ-P3uHm8tDJPNA==/base.apk!/lib/arm64-v8a, /system/lib64, /system/vendor/lib64]]
	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:134)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
	... 8 more

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30120190
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: Sony
Device: G8142
Model: G8142
Id: 47.2.A.10.107
Product: G8142

************ FIRMWARE ************
SDK: 28
Release: 9
Incremental: 172320177

--- a/build.gradle
+++ b/build.gradle
@@ -47,9 +47,6 @@ configurations {
     ktlint
 
     all {
-        exclude group: 'com.google.firebase', module: 'firebase-core'
-        exclude group: 'com.google.firebase', module: 'firebase-analytics'
-        exclude group: 'com.google.firebase', module: 'firebase-measurement-connector'
         exclude group: 'org.jetbrains', module: 'annotations-java5' // via prism4j, already using annotations explicitly
 
         // check for updates every build

--- a/gplay.gradle
+++ b/gplay.gradle
@@ -1,3 +1,3 @@
 dependencies {
-    implementation "com.google.firebase:firebase-messaging:20.1.3"
+    implementation "com.google.firebase:firebase-messaging:20.2.4"
 }

--- a/lint.xml
+++ b/lint.xml
@@ -62,5 +62,6 @@
         <ignore path="**/jetified-butterknife-runtime-10.**/**/lint.jar" />
         <ignore path="**/jetified-dagger-lint-aar-2.29.**/**/lint.jar" />
         <ignore path="**/jetified-annotation-experimental-1.**/**/lint.jar" />
+        <ignore path="**/jetified-firebase-installations**/**/lint.jar" />
     </issue>
 </lint>

--- a/src/gplay/AndroidManifest.xml
+++ b/src/gplay/AndroidManifest.xml
@@ -30,10 +30,13 @@
 
         <meta-data
             android:name="firebase_analytics_collection_deactivated"
-            android:value="true"/>
+            android:value="true" />
         <meta-data
             android:name="google_analytics_adid_collection_enabled"
-            android:value="false"/>
+            android:value="false" />
+        <meta-data
+            android:name="google_analytics_ssaid_collection_enabled"
+            android:value="false" />
 
         <activity
             android:name=".authentication.ModifiedAuthenticatorActivity"

